mautic/mautic
# Fix datetime contact value campaign condition

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



#### Steps to test this PR:

If you use datetime field to contact value campaign condition it was not compare properly. This PR fixed it


1. Change timezone in configuration to Europe/paris
2. Create datetime field 
3. Create contact and set datetime If today is 28.10, then set 27.10. 2021 00:00
4. Create campaign with condition field value with date -1 day

![image](https://user-images.githubusercontent.com/462477/139283015-4cb24f96-3896-4161-97af-a3e7bd038719.png)

5. Trigger campaign
6. Before contact was not true for condition because we compare datetime (yyyy-mm-dd h:i:s) with date (yyy-mm-dd). 
7. After PR should work  properly




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10560"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
